[dynamo][logger] Log graph break on Unsupported bytecodes

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122684

 This would have saved me a few hours while debugging an internal model. We could not support a LOAD_ATTR bytecode, because it was a property, and the inlining failed due to skip. Since LOAD_ATTR does not support continuation function, we would fallback to eager for the whole frame aka skip. But, we should also log this as graph break. This PR does it. 

Bonus - removes skip from a test.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

anijain2305 committed:
[dynamo][logger] Log graph break on Unsupported bytecodes

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"


 This would have saved me a few hours while debugging an internal model. We could not support a LOAD_ATTR bytecode, because it was a property, and the inlining failed due to skip. Since LOAD_ATTR does not support continuation function, we would fallback to eager for the whole frame aka skip. But, we should also log this as graph break. This PR does it. 

Bonus - removes skip from a test.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"


 This would have saved me a few hours while debugging an internal model. We could not support a LOAD_ATTR bytecode, because it was a property, and the inlining failed due to skip. Since LOAD_ATTR does not support continuation function, we would fallback to eager for the whole frame aka skip. But, we should also log this as graph break. This PR does it. 

Bonus - removes skip from a test.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

ezyang commented on torch/_dynamo/convert_frame.py:
```diff
@@ -696,6 +698,18 @@ def count_args(code):
+                        "Graph break: skip: from user code at:\n%s",
+                        user_stack_formatted,
+                        exc_info=True,
+                    )
```
I feel this is not the right place to do the catch. We are still propagating the exception upwards. The graph break log should happen when we suppress the exception.

anijain2305 replied:
The trace_id is not there available there. Its outside the _compile region.

anijain2305 replied:
This is the place where we propagate the exception upwards. I tried to put the log here, but the trace_id is not available there. So, we dont see prefixes like [0/0]

https://github.com/pytorch/pytorch/blob/852111e1c20c38eb260b3975ff7d90e511e7d93f/torch/_dynamo/convert_frame.py#L807-L841

anijain2305 replied:
Sorry for the spam - But I took your suggestion. Just stashing the compile_id to be used later on.

ezyang replied:
My counterproposal is that the error suppression should be moved inwards until the point where tracing id is set, OR the setting of tracing id is moved outwards so that it is coincident with this.

But I'm also... skeptical that this is the true Unsupported handler block? If it was, shouldn't you have gotten an INFO log for the exception anyways?

anijain2305 replied:
It was on info. But if you search for graph_break, you won't find it.

ezyang replied:
Too much logging :P

anijain2305 replied:
There is definitely duplication. I feel a usual reason of perf tanking is excessive graph break. So, just searching for `__graph_break` should give a good indication of what happened between a good and bad run.

I understand we can search for `WON'T CONVERT` in addition to `__graph_break` as a habit.

Hmm ... lets discuss offline.

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"


 This would have saved me a few hours while debugging an internal model. We could not support a LOAD_ATTR bytecode, because it was a property, and the inlining failed due to skip. Since LOAD_ATTR does not support continuation function, we would fallback to eager for the whole frame aka skip. But, we should also log this as graph break. This PR does it. 

Bonus - removes skip from a test.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"


 This would have saved me a few hours while debugging an internal model. We could not support a LOAD_ATTR bytecode, because it was a property, and the inlining failed due to skip. Since LOAD_ATTR does not support continuation function, we would fallback to eager for the whole frame aka skip. But, we should also log this as graph break. This PR does it. 

Bonus - removes skip from a test.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"


 This would have saved me a few hours while debugging an internal model. We could not support a LOAD_ATTR bytecode, because it was a property, and the inlining failed due to skip. Since LOAD_ATTR does not support continuation function, we would fallback to eager for the whole frame aka skip. But, we should also log this as graph break. This PR does it. 

Bonus - removes skip from a test.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"


 This would have saved me a few hours while debugging an internal model. We could not support a LOAD_ATTR bytecode, because it was a property, and the inlining failed due to skip. Since LOAD_ATTR does not support continuation function, we would fallback to eager for the whole frame aka skip. But, we should also log this as graph break. This PR does it. 

Bonus - removes skip from a test.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"


 This would have saved me a few hours while debugging an internal model. We could not support a LOAD_ATTR bytecode, because it was a property, and the inlining failed due to skip. Since LOAD_ATTR does not support continuation function, we would fallback to eager for the whole frame aka skip. But, we should also log this as graph break. This PR does it. 

Bonus - removes skip from a test.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][logger] Log graph break on Unsupported bytecodes"


 This would have saved me a few hours while debugging an internal model. We could not support a LOAD_ATTR bytecode, because it was a property, and the inlining failed due to skip. Since LOAD_ATTR does not support continuation function, we would fallback to eager for the whole frame aka skip. But, we should also log this as graph break. This PR does it. 

Bonus - removes skip from a test.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot rebase -b main

anijain2305 committed:
Update

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot merge
